frontend/app/src/components/accounts/balances/NonFungibleBalances.vue

```diff
@@ -115,31 +203,32 @@ defineProps({
   }
 });
 
-const edit: Ref<NonFungibleBalance | null> = ref(null);
-const confirmDelete: Ref<NonFungibleBalance | null> = ref(null);
-
 const balancesStore = useNonFungibleBalancesStore();
-const { nonFungibleTotalValue, fetchNonFungibleBalances } = balancesStore;
-const { nonFungibleBalances } = storeToRefs(balancesStore);
+const { fetchNonFungibleBalances: fetch, updateRequestPayload } = balancesStore;
+const { balances } = storeToRefs(balancesStore);
 const { currencySymbol } = storeToRefs(useGeneralSettingsStore());
-const total = nonFungibleTotalValue();
+const options: Ref<PaginationOptions | null> = ref(null);
+
 const { tc } = useI18n();
 const { notify } = useNotifications();
 
+const edit: Ref<NonFungibleBalance | null> = ref(null);
+const confirmDelete: Ref<NonFungibleBalance | null> = ref(null);
+
 const deleteAsset = computed(() => {
   const balance = get(confirmDelete);
   return !balance ? '' : balance.name ?? balance.id;
 });
 
-const mappedBalances = computed(() => {
-  return get(nonFungibleBalances).map(balance => {
-    return {
-      ...balance,
-      imageUrl: balance.imageUrl || '/assets/images/placeholder.svg',
-      isVideo: isVideo(balance.imageUrl)
-    };
-  });
-});
+const { data, limit, found, total, totalUsdValue } =
+  getCollectionData<NonFungibleBalance>(
+    balances as Ref<Collection<NonFungibleBalance>>
+  );
+
+const { itemLength } = setupEntryLimit(limit, found, total);
+
+const selected: Ref<NonFungibleBalance[]> = ref([]);
+const ignoredAssetsHandling = ref<IgnoredAssetsHandlingType>('exclude');
```

Comment:
I also mentioned it below, but it seems that we have some duplication on these server pagination table components, maybe we can create a generic reusable component to reduce the duplication